adapter/src/dbt/adapters/fal/teleport_support/duckdb.py

```diff
@@ -23,17 +25,35 @@ def teleport_from_external_storage(self, relation: BaseRelation, relation_path:
         url = teleport_info.build_url(relation_path)
 
         with self._db_adapter.connection_named('teleport:copy_from'):
-            rendered_macro = self._db_adapter.execute_macro('duckdb__copy_from', kwargs={'relation': relation, 'url': url})
+            if self.credentials.type == TeleportTypeEnum.REMOTE_S3:
+                # Putting this in __init__ didn't work, looks like it has to be done with each new connection
+                self._setup_s3()
+
+            macro_dict = {
+                's3': 'duckdb__copy_from_s3',
+                'local': 'duckdb__copy_from'
+            }
+
+            rendered_macro = self._db_adapter.execute_macro(
+                macro_dict[self.credentials.type.value],
+                kwargs={'relation': relation, 'url': url})
             self._db_adapter.execute(rendered_macro)
 
     def teleport_to_external_storage(self, relation: BaseRelation, teleport_info: TeleportInfo):
         assert teleport_info.format == 'parquet', "duckdb only supports parquet format for Teleport"
-
         rel_path = teleport_info.build_relation_path(relation)
         url = teleport_info.build_url(rel_path)
-
         with self._db_adapter.connection_named('teleport:copy_to'):
+            if self.credentials.type == TeleportTypeEnum.REMOTE_S3:
+                self._setup_s3()
             rendered_macro = self._db_adapter.execute_macro('duckdb__copy_to', kwargs={'relation': relation, 'url': url})
             self._db_adapter.execute(rendered_macro)
 
         return rel_path
+
+    def _setup_s3(self):
+        self._db_adapter.execute("INSTALL httpfs")
```

Comment:
Can we do `INSTALL` on init? Maybe next to `self._db_adapter.execute("INSTALL 'parquet'")`